File: from-3101-to-3200/3195-find-the-minimum-area-to-cover-all-ones-i/solution.go
func minimumArea(grid [][]int) int {
    m, n := len(grid), len(grid[0])
    var firstRow, lastRow, firstCol, lastCol int

    foundOne := false
    for r := 0; r < m; r ++ {
        for c := 0; c < n; c ++ {
            if grid[r][c] == 1 {
                foundOne = true
                firstRow = r
                break
            }
        }
        if foundOne {
            break
        }
    }

    foundOne = false
    for r := m - 1; r >= 0; r -- {
        for c := 0; c < n; c ++ {
            if grid[r][c] == 1 {
                foundOne = true
                lastRow = r
                break
            }
        }
        if foundOne {
            break
        }
    }

    foundOne = false 
    for c := 0; c < n; c ++ {
        for r := firstRow; r <= lastRow; r ++ {
            if grid[r][c] == 1 {
                foundOne = true
                firstCol = c
                break
            }
        }
        if foundOne {
            break
        }
    }

    foundOne = false 
    for c := n - 1; c >= 0; c -- {
        for r := firstRow; r <= lastRow; r ++ {
            if grid[r][c] == 1 {
                foundOne = true
                lastCol = c
                break
            }
        }
        if foundOne {
            break
        }
    }

    return (lastRow - firstRow + 1) * (lastCol - firstCol + 1)
}

File: from-3101-to-3200/3195-find-the-minimum-area-to-cover-all-ones-i/solution.py
class Solution:
    def minimumArea(self, grid: List[List[int]]) -> int:
        m, n = len(grid), len(grid[0])

        foundOne = False
        for r in range(m):
            for c in range(n):
                if grid[r][c] == 1:
                    firstRow = r
                    foundOne = True
                    break
            if foundOne:
                break

        foundOne = False
        for r in range(m - 1, -1, -1):
            for c in range(n):
                if grid[r][c] == 1:
                    lastRow = r
                    foundOne = True
                    break
            if foundOne:
                break

        foundOne = False
        for c in range(n):
            for r in range(firstRow, lastRow + 1):
                if grid[r][c] == 1:
                    firstCol = c
                    foundOne = True
            if foundOne:
                break

        foundOne = False
        for c in range(n - 1, -1, -1):
            for r in range(firstRow, lastRow + 1):
                if grid[r][c] == 1:
                    lastCol = c
                    foundOne = True
            if foundOne:
                break

        return (lastRow - firstRow + 1) * (lastCol - firstCol + 1)